.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,17 @@
+name: Prepare Go Release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  prepare_release:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+        
+      - name: Prepare Go Publish
+        run: |
+              cd release
+              make prep-go
```

Comment:
This introduces an extra step, in creating an RC branch through the pipeline. But I think this is a bit redundant, since it automates the process no more than its manual counterpart.

For all intents and purposes, we can consider the branches that the 1PasswordSDKBot opens to be the release candidates. So we have two situations:

1. We want a full release, with a new core and generated code. In this situation:
* the GitLab pipeline should generate the code and new core (which it already does), and, additionally, change the version number in the target SDK.
* devs would review these changes, approve, and merging this MR to main would trigger the SDK release workflow (tagging and releasing)
* the release would have the body/description of the merge request that it originated from

2. We want to do a release that doesn't involve updates in the core. In this situation:
* after merging the relevant changes to main, the developers run (locally) a script on latest main, which checks out a new branch rc branch and bumps the version
* devs would review these changes and approve - Merging this MR to main would trigger the SDK release workflow (tagging and releasing)
* the releas would have the body/description of the merge request that it originated from